lib/puppet/parser/functions/dirname.rb

```diff
@@ -16,6 +16,10 @@ module Puppet::Parser::Functions
     unless arguments[0].is_a?(String)
       raise(Puppet::ParseError, 'dirname(): Requires string as argument')
     end
+    # undef is converted to an empty string ''
+    if arguments[0].empty?
+      raise(Puppet::ParseError, 'dirname(): Requires a none empty string as argument')
```

Comment:
can you update:

`none empty` to `non-empty` and I will get your change merged.